core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -344,7 +357,7 @@ public void write(byte[] b, int off, int len) throws IOException {
         writeBytes(b, off, len);
     }
 
-    public void writeStringArray(String[] array) throws IOException {
+    public void writeStringArray(String... array) throws IOException {
```

Comment:
was this necessary? I tend to think we should keep it as the others. What does the vararg buy us here?